spring-transaction/src/test/kotlin/org/jetbrains/exposed/spring/SpringTransactionManagerTest.kt

```diff
@@ -367,13 +367,34 @@ class SpringTransactionManagerTest {
         assertEquals(0, con1.closeCallCount)
     }
 
+    @Test
+    fun `transaction timeout`() {
+        val tm = SpringTransactionManager(ds1)
+        tm.executeAssert(initializeConnection = true, timeout = 1) {
+            assertEquals(1, TransactionManager.current().queryTimeout)
+        }
+    }
+
+    @Test
+    fun `transaction timeout propagation`() {
+        val tm = SpringTransactionManager(ds1)
+        tm.executeAssert(initializeConnection = true, timeout = 1) {
+            tm.executeAssert(initializeConnection = true, timeout = 2) {
+                assertEquals(1, TransactionManager.current().queryTimeout)
+            }
+            assertEquals(1, TransactionManager.current().queryTimeout)
+        }
+    }
+
```

Comment:
`spring-transaction` tests run on H2 only, so an H2 hack will be needed.

I'm not the biggest fan of this, but if it's just one test and it gives us peace of mind for this feature, an option is using [a recursive query](https://www.h2database.com/html/advanced.html?highlight=recursive&search=re#recursive_queries) to simulate a long-running query:
```kt
tm.executeAssert(initializeConnection = true, timeout = 1) {
    try {
        TransactionManager.current().exec(
           """
               WITH RECURSIVE T(N) AS (
               SELECT 1
               UNION ALL
               SELECT N+1 FROM T WHERE N<10000000
               )
               SELECT * FROM T;
            """.trimIndent(),
            explicitStatementType = StatementType.SELECT
        )
        fail("Should have thrown a timeout exception")
    } catch (cause: ExposedSQLException) {
        assertTrue { cause.cause is SQLTimeoutException }
    }
}
```

I also tried this in the `ExposedTransactionManagerTest.kt` file using the annotation `@Transactional(timeout = 1)` and it worked.
Do you think this is solid enough or do you think we need to try something in the spring boot tests?

Reply:
Since timeout has already been tested in Exposed statements and in Transactions, I thought it would be okay to check that the SpringTransactionManager using Transactions sets the value in Transactions correctly.

However, I think that timeout using recursion is also meaningful enough to test the actual DB, so I will add the test code you suggested. Thank you.